website_docs/_index.md

```diff
@@ -30,3 +30,162 @@ The current release can be found [here](https://github.com/open-telemetry/opente
+# see SDK#configure for customizing the setup
+OpenTelemetry::SDK.configure
+
+# To start a trace you need to get a Tracer from the TracerProvider
+tracer = OpenTelemetry.tracer_provider.tracer('my_app_or_gem', '0.1.0')
+```
+
+### Instrumentation
+
+#### Creating Spans
+
+```ruby
+  begin
+    span = tracer.start_span("foo", kind: :internal)
+    Trace.with_span(span) do |span, context|
+      # custom code here
+      pp "Hello World!"
+    end
+  rescue Exception => e
+    span&.record_exception(e)
+    span&.status = Status.error("Unhandled exception of type: #{e.class}")
+    raise e
+  ensure
+    span&.finish
+```
+
+```ruby
+# create a span
+tracer.in_span('foo') do |span|
+  # set an attribute
+  span.set_attribute('platform', 'osx')
+  # add an event
+  span.add_event('event in bar')
+  # create bar as child of foo
+  tracer.in_span('bar') do |child_span|
+    # inspect the span
+    pp child_span
+  end
+end
+```
+
+#### Attributes
+
+Attributes are keys and values that are applied as metadata to your spans and are useful for aggregating, filtering, and grouping traces. Attributes can be added at span creation, or at any other time during the lifecycle of a span before it has completed.
+
+```ruby
+# setting attributes at creation...
+tracer.in_span('foo', attributes: {  "hello" => "world", "some.number" => 1024, "tags" => [ "bugs", "won't fix" ] }, kind: :internal) do |span|
+
+  # ... and after creation
+  span.set_attribute("animals", ["elephant", "tiger"])
+
+  span.add_attributes({ "my.cool.attribute" => "a value", "my.first.name" => "Oscar" })
+end
+```
+
+#### Semantic Attributes
+
+Semantic Attributes are attributes that are defined by the OpenTelemetry Specification in order to provide a shared set of attribute keys across multiple languages, frameworks, and runtimes for common concepts like HTTP methods, status codes, user agents, and more. These attributes are available in the [Semantic Conventions gem](https://github.com/open-telemetry/opentelemetry-ruby/tree/main/semantic_conventions).
+
+Tracing semantic conventions can be found in [this document](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions)
+
+#### Events
+
+An event is a human-readable message on a span that represents "something happening" during it's lifetime. For example, imagine a function that requires exclusive access to a resource that is under a mutex. An event could be created at two points - once, when we try to gain access to the resource, and another when we acquire the mutex.
+
+```ruby
+span.add_event("Acquiring lock")
+if mutex.try_lock
+  span.add_event("Got lock, doing work...")
+  # some code here
+  span.add_event("Releasing lock")
+else
+  span.add_event("Lock already in use")
+end
+```
+
+A useful characteristic of events is that their timestamps are displayed as offsets from the beginning of the span, allowing you to easily see how much time elapsed between them.
```

Comment:
This is an artifact of a given backend - it isn't universally true.

Reply:
Copy/paste Golang code